### DISCUSSION TITLE
Multiple matches in a line printed individually (w/o --vimgrep)

### DISCUSSION THREAD

**Discussion by acamso** [BurntSushi/ripgrep#2270] [Q&A] [answered]
Hi all, thanks for the _excellent_ tool!

I was wondering if there was a way to enable the `--vimgrep` flag's functionality for:

> With this option, a line with more than one match will be printed more than once.

.. without using the `--vimgrep` flag. I'm looking to use it in a terminal with the `--heading` flag. I read and search the man page, but didn't come across anything.

Cheers!

**Comment by BurntSushi** [comment]
Could you please explain your use case? What problem are you trying to solve?

Also, to clarify, does the `-o/--only-matching` flag work for your use case? (It is not specifically what you're asking for, but it's closely related.)

**Comment by acamso** [comment]
I'm setting up a global config for a new system. After reading the man page, it's just a personal preference of viewing the output this way outside of Vim. It's not too important - I'll update this if it's necessary. Thanks for the quick reply.  :+1:

**Comment by BurntSushi** [comment, marked as answer]
Aye yeah, there is no way to do what you want here. And I'm not inclined to add more flags to support it without a more compelling use case. The `--vimgrep` output mode is frankly weird, and it only exists because that's how vim wants things.
